front/lib/api/assistant/actions/database_query.ts

```diff
@@ -0,0 +1,250 @@
+  );
+
+  if (rawInputsRes.isErr()) {
+    return new Err(rawInputsRes.error);
+  }
+
+  if (rawInputsRes.isOk()) {
+    return new Ok(rawInputsRes.value);
+  }
+  return new Ok({});
+}
+
+/**
+ * Run the DatabaseQuery app.
+ */
+export async function* runDatabaseQueryApp({
+  auth,
+  configuration,
+  conversation,
+  userMessage,
+  agentMessage,
+}: {
+  auth: Authenticator;
+  configuration: AgentConfigurationType;
+  conversation: ConversationType;
+  userMessage: UserMessageType;
+  agentMessage: AgentMessageType;
+}): AsyncGenerator<DatabaseQueryRunErrorEvent | DatabaseQueryRunSuccessEvent> {
+  // Checking authorizations
+  const owner = auth.workspace();
+  if (!owner) {
+    throw new Error("Unexpected unauthenticated call to `runQueryDatabase`");
+  }
+  const c = configuration.action;
+  if (!isDatabaseQueryConfiguration(c)) {
+    throw new Error(
+      "Unexpected action configuration received in `runQueryDatabase`"
+    );
+  }
+  if (owner.sId !== c.dataSourceWorkspaceId) {
+    yield {
+      type: "database_query_run_error",
+      created: Date.now(),
+      configurationId: configuration.sId,
+      messageId: agentMessage.sId,
+      error: {
+        code: "database_query_parameters_generation_error",
+        message: "Cannot access the database linked to this action.",
+      },
+    };
+  }
+
+  // Generating inputs
+  const inputRes = await generateDatabaseQueryAppParams(
+    auth,
+    configuration,
+    conversation,
+    userMessage
+  );
+  if (inputRes.isErr()) {
+    yield {
+      type: "database_query_run_error",
+      created: Date.now(),
+      configurationId: configuration.sId,
+      messageId: agentMessage.sId,
+      error: {
+        code: "database_query_parameters_generation_error",
+        message: `Error generating parameters for database_query: ${inputRes.error.message}`,
+      },
+    };
+    return;
+  }
+  const input = inputRes.value;
+
+  // Generating configuration
+  const config = cloneBaseConfig(
+    DustProdActionRegistry["assistant-v2-query-database"].config
+  );
+  const database = {
+    workspace_id: c.dataSourceWorkspaceId,
+    data_source_id: c.dataSourceId,
+    database_id: c.databaseId,
+  };
+  config.DATABASE_SCHEMA = {
+    type: "database_schema",
+    database,
+  };
+  config.DATABASE = {
+    type: "database",
+    database,
+  };
+
+  // Running the app
+  const res = await runActionStreamed(
+    auth,
+    "assistant-v2-query-database",
+    config,
+    [{ question: input.question }]
+  );
+
+  if (res.isErr()) {
+    yield {
+      type: "database_query_run_error",
+      created: Date.now(),
+      configurationId: configuration.sId,
+      messageId: agentMessage.sId,
+      error: {
+        code: "dust_app_run_error",
+        message: `Error running DatabaseQuery app: ${res.error.message}`,
+      },
+    };
+    return;
+  }
+
+  const { eventStream } = res.value;
+  for await (const event of eventStream) {
+    if (event.type === "block_execution") {
```

Comment:
You're not catching run errors here so we'll loose the cause of error. Let's cago-cult retrieval.ts here.